client/reader/components/favicon/index.js

```diff
@@ -8,8 +8,8 @@ function Favicon( props ) {
 	const { site, className, size } = props;
 	const [ hasError, setError ] = useState( false );
 
-	// if loading error show W Gridicon
-	if ( hasError ) {
+	// if loading error or missing icon show W Gridicon
+	if ( hasError || site.site_icon === null ) {
```

Comment:
Just wondering if `site` will always have `site_icon` prop and whether we should use lodash's `get` to guard here but probably not necessary.

Reply:
At the moment the endpoint always returns it either `null` or `string`